### PR TITLE
Feature: One click connect to node with ssh

### LINKED ISSUE #1452: BatchLabs one click node connect
Have a one click connect button that use `~/.ssh/id_rsa.pub` if present and generate username and open a terminal

- [ ] Let user pick which terminal to open(default powershell on windows)
- [ ] Let user pick what is the username of the user to be created(Otherwise just random)

### PR DESCRIPTION
**Pull request by arinehouse**
fix #1452

### CODE REVIEW

File: src/client/core/terminal/terminal.service.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import * as cp from "child_process";
+
+const supportedTerminals = {
+    powershell: {
+        proc: "cmd.exe",
```
Comment (timotheeguerin) [inline src/client/core/terminal/terminal.service.ts]: `process`

File: ssh.exe.stackdump [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Stack trace:
```
Comment (timotheeguerin) [inline ssh.exe.stackdump]: ignore this file

File: app/services/batch-labs.service.ts [hunk 3]
```diff
@@ -57,6 +57,10 @@ export class BatchLabsService {
         return this.remote.send(IpcEvent.logoutAndLogin);
     }
 
+    public async launchApplication(args): Promise<any> {
```
[the remark this answers is not in the document]
Comment (arinehouse) [reply app/services/batch-labs.service.ts]: It's just an object. Do you want `public async launchApplication(args: any): Promise<any> {` or an actual interface definition up above?

**Comment by arinehouse** [comment]
Travis build is failing on this line:
```
$ ./scripts/travis/build-and-test.sh
/Users/travis/.travis/job_stages: line 78: ./scripts/travis/build-and-test.sh: Permission denied
The command "./scripts/travis/build-and-test.sh" exited with 126.
```

**Comment by timotheeguerin** [comment]
hhm so it seems like you changed the `build-and-test.sh` file. not sure how but it probably changed the execution rules on it `scripts/travis/build-and-test.sh 100755 → 100644`

Might need to update this as here https://stackoverflow.com/questions/42154912/permission-denied-for-build-sh-file

File: src/client/core/terminal/terminal.service.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+const supportedTerminals = {
+    powershell: {
+        process: "cmd.exe",
+        args: ["/c", "start", "powershell", "-NoExit", "-Command", "{command}"],
+    },
+    cmd: {
+        process: "cmd.exe",
+        args: ["/c", "start", "cmd", "/k", "{command}"],
+    },
+    linux: {
```
Comment (timotheeguerin) [inline src/client/core/terminal/terminal.service.ts]: here I was more thinking of listing all the linux terminals and depending on what the user choose get that.

File: app/components/node/connect/node-connect.component.spec.ts [hunk 5]
```diff
@@ -54,31 +111,36 @@ describe("NodeConnectComponent", () => {
...
         component = de.componentInstance;
-        component.pool = Fixtures.pool.create();
-        component.node = Fixtures.node.create();
+        testComponent.pool = Fixtures.pool.create();
+        testComponent.node = Fixtures.node.create();
         fixture.detectChanges();
     });
 
-    it("should propose to generate or specify credentials", () => {
+        it("should propose to generate or specify credentials, or connect in one click", () => {
```
Comment (timotheeguerin) [inline app/components/node/connect/node-connect.component.spec.ts]: tab? and other below

**Pull request merged**